src/repo/secret/infisical.go

```diff
@@ -0,0 +1,38 @@
+package secret
+
+import (
+	"sync"
+
+	infisical "github.com/infisical/go-sdk"
+)
+
+type Secret struct {
+	client infisical.InfisicalClientInterface
+	mu     sync.Mutex
+}
+
+func NewSecret(secret *Secret) (*Secret, error) {
+	secret.mu.Lock()
+	client := infisical.NewInfisicalClient(infisical.Config{
+		SiteUrl: "https://app.infisical.com",
+	})
+	// use INFISICAL_UNIVERSAL_AUTH_CLIENT_ID, INFISICAL_UNIVERSAL_AUTH_CLIENT_SECRET env
+	_, err := client.Auth().UniversalAuthLogin("", "")
```

Comment:
this function returns secret value, we must do the following:
- create field in the  `type Secret struct`
- create function `issueToken` or something, inside this func we save the token to the state
- before issuing the token if must check if the token is valid in order not to issue a token if it's valid. we can save time to live as `ttl time.Duration` and check if it's fresh enough
- get the token on every operation like `GetSecret`